### DISCUSSION TITLE
esp32 wifi: how to use ap mode and ap+station mode?

### DISCUSSION THREAD

**Discussion by zoujixing** [zephyrproject-rtos/zephyr#51157] [Q&A]
On V3.1 branch version;
I want to use esp32 wifi as ap mode and ap+station mode. Any suggestion or reference documents?

**Comment by stephanosio** [comment]
Please do not create issues (especially bug) for questions.

**Comment by rftafas** [comment]
Hi!

AFAIK, some people tested it to varying degrees of success using static IPs, with lots of dependence on the equipments they are interfacing too. They had to do lots of hacking and use things from our HAL directly. Provided that hal_espressif is not a public API, it should not be used for building applications or products: it can change. 

So, all in all, you won't find user documentation about AP mode. It is not supported.

It is even worst on this case. Besides being untested, there would be necessary lots of changes on the current driver to make it work and probably forcing more usage of hal_espressif directly. So, this not only is not supported, but probably won't work.

On a bright side, some discussions are being held and as soon as they get somewhere, we might start focusing on AP mode. PTAL:

1. DHCP server #38284
2. Zephyr Supplicant Layer  #34507 (we have one embedded on our driver, made transparent by using wifi manager)

Until that is more clear, there is a lot of other important work to do on other fronts.
